functions/Test-DbaPowerPlan.ps1

```diff
@@ -59,71 +59,116 @@ Test-DbaPowerPlan -ComputerName sqlserver2014a -Detailed
-				$powerplan.InstanceID = $powerplan.InstanceID.Split('{')[1].Split('}')[0]
-
-				if ($CustomPowerPlan.Length -gt 0)
+				$cimsession = New-CimSession -ComputerName $Computer -ErrorAction SilentlyContinue -Credential $Credential
+			}
+			
+			if ($null -eq $cimsession.id)
+			{
+				Write-Message -Level Verbose -Message "Creating CimSession on $computer over WSMan failed. Creating CimSession on $computer over DCom"
+				
+				if (!$Credential)
 				{
-					$bpPowerPlan.ElementName = $CustomPowerPlan
-					$bpPowerPlan.InstanceID = $( $powerplans | Where-Object {  $_.ElementName -eq $CustomPowerPlan }).InstanceID
+					$cimsession = New-CimSession -ComputerName $Computer -SessionOption $sessionoption -ErrorAction SilentlyContinue -Credential $Credential
 				}
 				else
 				{
-					$bpPowerPlan.ElementName =  $( $powerplans | Where-Object {  $_.InstanceID.Split('{')[1].Split('}')[0] -eq $bpPowerPlan.InstanceID }).ElementName
-					if ($null -eq $bpPowerplan.ElementName)
-					{
-						$bpPowerPlan.ElementName = "You do not have the high performance plan installed on this machine."
-					}
+					$cimsession = New-CimSession -ComputerName $Computer -SessionOption $sessionoption -ErrorAction SilentlyContinue
 				}
-
+			}
+			
+			if ($null -eq $cimsession.id)
+			{
+				Stop-Function -Message "Can't create CimSession on $computer"
+			}
+			
+			Write-Message -Level Verbose -Message "Getting Power Plan information from $Computer"
+			
+			try
+			{
+				$powerplans = Get-CimInstance -CimSession $cimsession -classname Win32_PowerPlan -Namespace "root\cimv2\power" -ErrorAction Stop | Select-Object ElementName, InstanceID, IsActive
 			}
 			catch
 			{
-				Write-Warning "Can't connect to WMI on $server"
-				return
+				if ($_.Exception -match "namespace")
+				{
+					Stop-Function -Message "Can't get Power Plan Info for $Computer. Unsupported operating system." -Continue -InnerErrorRecord $_
+				}
+				else
+				{
+					Stop-Function -Message "Can't get Power Plan Info for $Computer. Check logs for more details." -Continue -InnerErrorRecord $_
+				}
 			}
-
+			
+			$powerplan = $powerplans | Where-Object { $_.IsActive -eq 'True' } | Select-Object ElementName, InstanceID
+			$powerplan.InstanceID = $powerplan.InstanceID.Split('{')[1].Split('}')[0]
+			
+			if ($CustomPowerPlan.Length -gt 0)
+			{
+				$bpPowerPlan.ElementName = $CustomPowerPlan
+				$bpPowerPlan.InstanceID = $($powerplans | Where-Object { $_.ElementName -eq $CustomPowerPlan }).InstanceID
+			}
+			else
+			{
+				$bpPowerPlan.ElementName = $($powerplans | Where-Object { $_.InstanceID.Split('{')[1].Split('}')[0] -eq $bpPowerPlan.InstanceID }).ElementName
+				if ($null -eq $bpPowerplan.ElementName)
+				{
+					$bpPowerPlan.ElementName = "You do not have the high performance plan installed on this machine."
+				}
+			}
+			
 			Write-Verbose "Recommended GUID is $($bpPowerPlan.InstanceID) and you have $($powerplan.InstanceID)"
```

Comment:
Should be:
`Write-Message -Level Verbose -Message "Recommended GUID is $($bpPowerPlan.InstanceID) and you have $($powerplan.InstanceID)"`